scenarios/aws/kindvm/run.go

```diff
@@ -154,3 +170,36 @@ agents:
 
 	return nil
 }
+
+func ConfigureECRCredentials(e resAws.Environment, vm *remote.Host, arch os.Architecture, opts ...pulumi.ResourceOption) (*goremote.Command, error) {
+	architecture := "x86_64"
+	if arch == os.ARM64Arch {
+		architecture = "aarch64"
+	}
+
+	unzipInstallCommand, err := vm.OS.PackageManager().Ensure("unzip", nil, "", opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	awsCliInstallCommand, err := vm.OS.Runner().Command(
+		e.CommonNamer.ResourceName("aws-cli-install"),
+		&command.Args{
+			Create: pulumi.Sprintf("command -v aws || curl 'https://awscli.amazonaws.com/awscli-exe-linux-%s.zip' -o 'awscliv2.zip' && unzip awscliv2.zip && sudo ./aws/install", architecture),
+		},
+		utils.PulumiDependsOn(unzipInstallCommand),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	ecrLoginCommand, err := vm.OS.Runner().Command(
+		e.CommonNamer.ResourceName("ecr-login"),
+		&command.Args{
+			Create: pulumi.Sprintf("aws ecr get-login-password | docker --config /tmp/kind-config login  --username AWS --password-stdin %s", e.CloudProviderEnvironment.InternalRegistry()),
+		},
+		utils.PulumiDependsOn(awsCliInstallCommand),
+	)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Oh so this is generating the `config.json` that is mounted to kind?

Reply:
Yep we retrieve the password from `aws` cli and we use it in the `docker login` command that will generate an auth token stored in the config file specified with `--config`
Note: I verified that the auth token generated expires when the password given by AWS does. The token should remaind valid for 12 hours